### PR TITLE
Fix Windows build

### LINKED ISSUE #67: Windows CI not succeeding
## Description
Latest `main` does not pass CI on Windows
 - This is follow up from #62 
See example build:
https://ci.ros2.org/job/ci_windows/18107/console

### Expected Behavior
Windows CI shall pass green without errors and warnings

## Additional context
Build fails with error message:
```
CMake Error at C:/ci/ws/install/share/mcap_vendor/cmake/mcapExport.cmake:85 (message):
16:13:36 CMake Error at C:/ci/ws/install/share/mcap_vendor/cmake/mcapExport.cmake:85 (message):
16:13:36   The imported target "mcap_vendor::mcap" references the file
16:13:36      "C:/ci/ws/install/lib/mcap.lib"
16:13:36   but this file does not exist.  Possible reasons include:
16:13:36   * The file was deleted, renamed, or moved to another location.
16:13:36   * An install or uninstall procedure did not complete successfully.
16:13:36   * The installation package was faulty and contained
16:13:36      "C:/ci/ws/install/share/mcap_vendor/cmake/mcapExport.cmake"
16:13:36   but not all the files it references.
16:13:36 Call Stack (most recent call first):
16:13:36   C:/ci/ws/install/share/mcap_vendor/cmake/ament_cmake_export_targets-extras.cmake:9 (include)
16:13:36   C:/ci/ws/install/share/mcap_vendor/cmake/mcap_vendorConfig.cmake:41 (include)
16:13:36   CMakeLists.txt:23 (find_package)
```
From the log file It looks like `mcap.lib` hasn't been built during the build process for some unknown reasons.
Also from logs it looks like it was built `mcap.dll` but not `mcap.lib`.

### PR DESCRIPTION
**Pull request by emersonknapp**
Depends on foxglove/mcap#711
Depends on foxglove/mcap#713
Fixes #67

### CODE REVIEW

File: mcap_vendor/CMakeLists.txt [hunk 1]
```diff
@@ -21,7 +21,8 @@ macro(build_mcap_vendor)
   include(FetchContent)
   fetchcontent_declare(mcap
     GIT_REPOSITORY https://github.com/foxglove/mcap.git
-    GIT_TAG 066f03b9b8444ffc1134e852e3419e4a4c0bef85 # releases/cpp/v0.6.0
+    # GIT_TAG 066f03b9b8444ffc1134e852e3419e4a4c0bef85 # releases/cpp/v0.6.0
```
Comment (emersonknapp) [inline mcap_vendor/CMakeLists.txt]: TODO: Undo this change and target updated mcap release hash once the Windows-export change is merged and released

**Comment by emersonknapp** [comment]
Gist: https://gist.githubusercontent.com/emersonknapp/971c0d3b5a93187a3fae8749cf4d59b8/raw/f3bf6eb8a0129a13415cefdf2c5c73d1d9ec08cc/ros2.repos
BUILD args:  --packages-above-and-dependencies rosbag2_storage_mcap mcap_vendor
TEST args:  --packages-above rosbag2_storage_mcap mcap_vendor
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/11088
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17605/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12146/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18175/)

File: rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* This header must be included by all rclcpp headers which declare symbols
+ * which are defined in the rclcpp library. When not building the rclcpp
+ * library, i.e. when using the headers in other package's code, the contents
+ * of this header change the visibility of certain symbols which the rclcpp
+ * library cannot have, but the consuming code must have inorder to link.
+ */
```
Comment (jtbandes) [inline rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp]: is this comment outdated?

Comment (emersonknapp) [reply rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp]: Yes, good catch. It's a copy of a copy

File: rosbag2_storage_mcap/src/mcap_storage.cpp [hunk 3]
```diff
@@ -47,7 +47,7 @@
 #include <utility>
 #include <vector>
 #ifdef ROSBAG2_STORAGE_MCAP_HAS_STORAGE_FILTER_TOPIC_REGEX
-#include <regex>
+  #include <regex>
```
Comment (james-rms) [inline rosbag2_storage_mcap/src/mcap_storage.cpp:50]: does it work to include spaces before the #?

File: mcap_vendor/CMakeLists.txt (hunk 1 above)
Comment (jhurliman) [reply mcap_vendor/CMakeLists.txt]: https://github.com/foxglove/mcap/pull/711 has merged, did the release happen?

Comment (emersonknapp) [reply mcap_vendor/CMakeLists.txt]: Was waiting on followup foxglove/mcap#713. Release happened just now, and this is updated

File: rosbag2_storage_mcap/src/mcap_storage.cpp (hunk 3 above)
Comment (emersonknapp) [reply rosbag2_storage_mcap/src/mcap_storage.cpp:50]: See `.clang-format` https://github.com/ros-tooling/rosbag2_storage_mcap/pull/73/files#diff-f064a7203feb3ead7ec4759c19472ef142aaaf2c2fcd526ee719575d7b1e95a0R20 - this package uses `BeforeHash` setting to better match the existing rosbag2 source style

**Comment by emersonknapp** [comment]
Gist: https://gist.githubusercontent.com/emersonknapp/338b589df9c52254f3cd5436a719cfac/raw/f3bf6eb8a0129a13415cefdf2c5c73d1d9ec08cc/ros2.repos
BUILD args:  --packages-above-and-dependencies rosbag2_storage_mcap mcap_vendor
TEST args:  --packages-above rosbag2_storage_mcap mcap_vendor
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/11130
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17645/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12200/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18219/)

**Comment by emersonknapp** [comment]
Setting `_CRT_SECURE_NO_WARNINGS`

Windows [Build Status](https://ci.ros2.org/job/ci_windows/18221/)

**Review by james-rms — APPROVED** [review]
